generator/src/main/java/com/scottlogic/deg/generator/restrictions/ParsedDateGranularity.java

```diff
@@ -0,0 +1,27 @@
+package com.scottlogic.deg.generator.restrictions;
+
+import com.scottlogic.deg.generator.generation.fieldvaluesources.datetime.Timescale;
+import com.scottlogic.deg.generator.utils.NumberUtils;
+
+import java.math.BigDecimal;
+import java.sql.Time;
+
+/**
+ * Granularity expressions could be interpreted differently depending on other constraints on a field (eg, type constraints),
+ * so we store all possible parsings in this class, ready to make a GranularityRestrictions object
```

Comment:
Do we? I can't see GranularityRestrictions anywhere else. Looks like an irrelevant comment for an old implementation. Rewrite this or delete it.